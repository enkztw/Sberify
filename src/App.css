html {
  font-family: 'Roboto', sans-serif;
  font-size: 2vw;
}

.App-link {
  color: #19bb4f;
  font-size: 4rem;
}

.App-text {
  color: #000000;
}

.container {
  display: flex;
  justify-content: center;
  align-items: flex-start;

  background-color: #000000;
}

.artist {
  width: 100%;
}

.artist__header {
  position: relative;

  display: block;
  height: 10rem;
  width: 100%;

  background-image: url("https://images.genius.com/11d94a5c2954e02819f8338ef30bdf6c.1000x667x1.jpg");
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
}

.artist__header::before {
  position: absolute;

  content: '';

  width: 100%;
  height: 100%;

  background: linear-gradient(to bottom,rgba(0,0,0,.1) 70%,rgba(0,0,0,.7) 93%,#000 99%);

  left: 0;
  top: 0;
}

.artist__info {
  width: 60vw;
  
  display: flex;
  flex-wrap: wrap;
  outline: 1px solid red;

  margin: 0 auto;
}

.artist__block {
  position: relative;

  width: 50%;
  height: 100%;

  display: flex;
  flex-direction: column;
  align-items: center;
}

.artist__image {
  position: absolute;

  width: 8rem;
  height: 8rem;

  box-sizing: border-box;
  border: 0.1rem solid #19bb4f;
  border-radius: 50%;

  top: -7rem;
}

.artist__name {
  font-size: 0.75rem;

  color: #ffffff;
  margin: 1.5rem 0 0;
  text-transform: uppercase;

  border-bottom: 0.05rem solid #19bb4f;
}

.artist__socials {
  list-style: none;
  padding: 0;
}

.artist__description {
  box-sizing: border-box;

  width: 25vw;
  padding: 1rem;

  border-radius: 0.15rem;
  background-color: #ffffff;

  margin: 0.5rem 0;
}

.artist__description p {
  font-size: 0.5rem;
  
  margin: 0;
}

.artist__description p:not(:last-of-type) {
  margin-bottom: 0.5rem;
}