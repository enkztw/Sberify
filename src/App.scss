@import "scss/preloader";
@import "scss/not-found";
@import "scss/album";
@import "scss/songs";



html {
  font-size: 2.75vw;
  font-family: "Roboto", sans-serif;
}

ul,
p,
h1,
h2,
h3,
h4 {
  margin: 0;
  padding: 0;
}

.container {
  display: flex;
  justify-content: center;
  align-items: flex-start;

  background-color: #000000;
}

.artist {
  width: 100%;
  min-height: 100vh;
}

.artist__header {
  position: relative;

  display: block;
  width: 100%;
  height: 10rem;

  background-image: url("https://images.genius.com/11d94a5c2954e02819f8338ef30bdf6c.1000x667x1.jpg");
  background-repeat: no-repeat;
  background-position: center 30%;
  background-size: cover;

  &::before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;

    width: 100%;
    height: 100%;

    background: linear-gradient(to bottom, rgba(0, 0, 0, 0.1) 70%, rgba(0, 0, 0, 0.7) 93%, #000000 99%);
  }
}

.artist__info {
  display: flex;
  flex-wrap: wrap;
  margin: 0 auto;
  padding: 0 7% 5%;
}

.artist__block {
  position: relative;

  display: flex;
  flex-direction: column;
  align-items: center;
  width: 50%;
  height: 100%;
}

.artist__image {
  position: absolute;
  top: -7rem;
  z-index: 2;

  box-sizing: border-box;
  width: 8rem;
  height: 8rem;

  border: 0.1rem solid #19bb4f;
  border-radius: 50%;
}

.artist__name {
  position: relative;
  z-index: 1;

  margin: 1.75rem 0 0;

  font-size: 0.75rem;
  line-height: 1rem;
  color: #ffffff;
  text-transform: uppercase;

  border-bottom: 0.05rem solid #19bb4f;

  &::before {
    content: "";
    position: absolute;
    top: 50%;
    left: 50%;
    z-index: -1;

    box-shadow: 0 100px 500px 125px rgba(25, 187, 79, 0.4);
  }
}

.artist__socials {
  position: relative;
  z-index: 1;

  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  margin: 0.6rem 0;
  padding: 0;

  list-style: none;

  background-color: transparent;


  li {
    width: 1rem;
    height: 1rem;

    &:not(:last-of-type) {
      margin-right: 0.5rem;
    }
  }
}

.artist__description-wrapper {
  position: relative;
  z-index: 1;

  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  width: 12rem;
  max-height: 12rem;
  margin: 1rem 0 0;
  padding: 1rem;

  background-color: #f5f5f6;
  border-radius: 0.25rem;
}

.artist__description {
  position: relative;
  z-index: 1;

  width: 100%;
  max-height: 10rem;
  overflow-y: scroll;

  border-radius: 0.25rem;

  &::-webkit-scrollbar {
    display: none;
  }

  p {
    margin: 0;

    font-size: 0.5rem;
    text-align: right;

    &:not(:last-of-type) {
      margin-bottom: 0.75rem;
    }

    &:nth-of-type(2n) {
      text-align: left;
    }
  }
}

.artist__block:last-of-type {
  article {
    &:not(:last-of-type) {
      margin-bottom: 0.75rem;
    }

    >h4 {
      position: relative;
      z-index: 2;

      width: 9rem;

      font-size: 0.35rem;
      line-height: 1rem;
      color: #ffffff;
      text-transform: uppercase;
    }
  }
}

.artist__albums {
  ul {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    width: 9rem;

    list-style: none;

    .album {
      width: 4rem;

      >a {
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
      }

      &:nth-of-type(2n) {
        margin-left: 1rem;
      }

      &:hover img {
        transform: scale(1.05);

        transition-duration: 0.15s;
        transition-property: transform;
      }

      img {
        position: relative;
        z-index: 1;

        display: block;
        width: 4rem;
        height: 4rem;

        box-shadow: 0 0 200px 50px rgba(25, 187, 79, 0.525);

        transition-duration: 0.15s;
        transition-property: transform;
      }

      h3 {
        position: relative;
        z-index: 2;

        margin: 0.25rem 0 0;

        font-weight: 400;
        font-size: 0.35rem;
        text-align: center;
        color: #ffffff;
      }

      span {
        position: relative;
        z-index: 2;

        margin-top: 0.1rem;

        font-size: 0.3rem;
        color: #19bb4f;
      }

      &:nth-of-type(n+3) {
        margin-top: 0.5rem;
      }
    }
  }
}

iframe {
  position: relative;
  z-index: 1;

  width: 12rem;
  height: 6.75rem;
  margin: 0.75rem 0;
}