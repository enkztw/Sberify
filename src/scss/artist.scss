.artist {
    &__info {
        display: flex;
        flex-wrap: wrap;
        margin: 0 auto;
        padding: 0 5% 5%;
    }

    &__block {
        padding: 0 1.5rem;
    }
}