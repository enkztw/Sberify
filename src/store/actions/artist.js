import * as types from '../action-types'
import axios from 'axios'

export const initArtist = payload => dispatch => {
  dispatch({
    type: types.SHOW_PRELOADER
  })

  axios
    .get(`/api/artist/${payload}`)
    .then(response => {
      dispatch({
        type: types.INIT_ARTIST,
        data: response.data
      })
    })
    .then(() => {
      dispatch({
        type: types.HIDE_PRELOADER
      })
    })
}

export const addArtist = payload => (dispatch, getState) => {
  dispatch({
    type: types.SHOW_PRELOADER
  })

  axios
    .put(`/api/artist/${payload}`)
    .then(response => {
      // dispatch({
      //   type: types.INIT_ARTIST,
      //   data: response.data
      // })
      console.log(response)
      // Dispatch redirect
    })
    .then(() => {
      dispatch({
        type: types.HIDE_PRELOADER
      })
    })
}
