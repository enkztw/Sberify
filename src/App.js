import React from 'react'

import Artist from './components/artist'

import './App.scss'

function App() {
  return <Artist />
}

export default App
