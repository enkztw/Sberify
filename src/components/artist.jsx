import React from "react";
import Lottie from "lottie-react-web";
import Social from "./social";
import Play from "./play";
import twitter from "../lottie/twitter";
import facebook from "../lottie/facebook";
import instagram from "../lottie/instagram";
import play from "../lottie/play";
import "../App.scss";

export default class Artist extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isStopped: true,
      isPaused: false
    };

    this.lottie = React.createRef();
  }

  render() {
    return (
      <div className="container">
        <section className="artist">
          <div className="artist__header"></div>
          <div className="artist__info">
            <div className="artist__block">
              <img
                src="https://images.genius.com/9b28279cfa5bbfee31181d32d6c5901a.750x750x1.jpg"
                alt="Artist"
                className="artist__image"
              />
              <h2 className="artist__name">Architects</h2>
              <ul className="artist__socials">
                <Social data={twitter} isLooped={true} />
                <Social data={facebook} isLooped={true} />
                <Social data={instagram} isLooped={false} />
              </ul>
              <div className="artist__description">
                <p>
                  Architects was formed in Brighton, England in 2005. They are
                  known for their blend of math rock, metalcore, djent, and
                  symphonic metal with sociopolitically charged lyrics. They
                  have released eight studio albums as of 2018’s Holy Hell.
                </p>
                <p>
                  In August 2016, founding member and guitarist Tom Searle
                  passed away from cancer. Drummer Dan Searle posted on Facebook
                  that the future of the band remained uncertain after the
                  conclusion of the All Our Gods… tour.
                </p>
                <p>
                  To pretend that Tom wasn’t at the heart of everything that the
                  band created would be to show a complete lack of respect to
                  the amazing talent that he was. The band will never be the
                  same and there is simply no denying it
                </p>
              </div>
              <iframe
                id="ytplayer"
                type="text/html"
                width="640"
                height="360"
                title="Clip"
                src="https://www.youtube.com/embed/WqRYBWyvbRo?autoplay=1&mute=1"
                frameBorder="0"
              />
            </div>
            <div className="artist__block">
              <article className="artist__songs">
                <h4>Popular Songs</h4>
                <ul>
                  <li className="song">
                    <Play data={play} />
                    <img
                      src="https://t2.genius.com/unsafe/220x220/https%3A%2F%2Fimages.genius.com%2F24107992fb59cf17720ec63b6677ea95.1000x1000x1.jpg"
                      alt="Song"
                    />
                    <div className="song__info">
                      <h3>Doomsday</h3>
                      <h4>Architects</h4>
                    </div>
                  </li>
                  <li className="song">
                    <Play data={play} />
                    <img
                      src="https://t2.genius.com/unsafe/220x220/https%3A%2F%2Fimages.genius.com%2F50300f39b82906f25d376f7f3ef32551.1000x1000x1.jpg"
                      alt="Song"
                    />
                    <div className="song__info">
                      <h3>Gone with the Wind</h3>
                      <h4>Architects</h4>
                    </div>
                  </li>
                  <li className="song">
                    <Play data={play} />
                    <img
                      src="https://t2.genius.com/unsafe/220x0/https%3A%2F%2Fimages.genius.com%2Fba1d44c4be5036977fcba61bb918572d.1000x1000x1.jpg"
                      alt="Song"
                    />
                    <div className="song__info">
                      <h3>Gravedigger</h3>
                      <h4>Architects</h4>
                    </div>
                  </li>
                </ul>
              </article>
              <article className="artist__albums">
                <h4>Latest Albums</h4>
                <ul>
                  <li className="album">
                    <img
                      src="https://t2.genius.com/unsafe/300x0/https%3A%2F%2Fimages.genius.com%2Ff551b358be75a07ae87fbbff22e1a93f.640x640x1.jpg"
                      alt="Album"
                    />
                    <h3>Spotify Singles</h3>
                    <span>2019</span>
                  </li>
                  <li className="album">
                    <img
                      src="https://images.genius.com/9bfc1be88070994ad4fbe90140f50cdb.300x300x1.jpg"
                      alt="Album"
                    />
                    <h3>Holy Hell</h3>
                    <span>2018</span>
                  </li>
                  <li className="album">
                    <img
                      src="https://t2.genius.com/unsafe/150x0/https%3A%2F%2Fimages.genius.com%2F0e1f84ddb67d4e5fd1fe45e9511206ac.300x300x1.jpg"
                      alt="Album"
                    />
                    <h3>Daybreaker</h3>
                    <span>2012</span>
                  </li>
                  <li className="album">
                    <img
                      src="https://images.genius.com/50300f39b82906f25d376f7f3ef32551.300x300x1.jpg"
                      alt="Album"
                    />
                    <h3>All Our Gods Have Abandoned Us</h3>
                    <span>2016</span>
                  </li>
                  <li className="album">
                    <img
                      src="https://t2.genius.com/unsafe/150x0/https%3A%2F%2Fimages.genius.com%2Fba1d44c4be5036977fcba61bb918572d.300x300x1.jpg"
                      alt="Album"
                    />
                    <h3>Lost Forever // Lost Together</h3>
                    <span>2014</span>
                  </li>
                  <li className="album">
                    <img
                      src="https://t2.genius.com/unsafe/300x0/https%3A%2F%2Fimages.genius.com%2F88247065a75bd0e82f3eba3c8aa79d76.1000x1000x1.jpg"
                      alt="Album"
                    />
                    <h3>The Here and Now</h3>
                    <span>2011</span>
                  </li>
                </ul>
              </article>
            </div>
          </div>
        </section>
      </div>
    );
  }
}
