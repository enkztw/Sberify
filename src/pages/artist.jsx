import React from 'react'

import { Helmet } from 'react-helmet'
import { connect } from 'react-redux'
import { actions } from '../store'
import { Redirect } from 'react-router-dom'

import Block from '../components/block'
import Social from '../components/social'
import Songs from '../components/songs'
import Albums from '../components/album/list'
import Video from '../components/video'

import twitter from '../lottie/twitter'
import facebook from '../lottie/facebook'
import instagram from '../lottie/instagram'

import '../App.scss'

const socialsMap = {
  twitter: {
    isLooped: true,
    data: twitter
  },
  facebook: {
    isLooped: true,
    data: facebook
  },
  instagram: {
    isLooped: false,
    data: instagram
  }
}

export class Artist extends React.Component {
  constructor(props) {
    super(props)

    this.state = {}

    console.log(window.navigator.userAgent)
  }

  componentDidMount() {
    this.props.clearArtists()
    this.props.initArtist(this.props.match.params.name)
  }

  render() {
    if (this.props.isFound === false) {
      return <Redirect to="/404/" />
    }

    return (
      <React.Fragment>
        <Helmet>
          <title>{this.props.name}</title>
          <meta property="og:site_name" content="Sberify Music App" />
          <meta property="og:url" content={window.location.href} />
          <meta property="og:type" content="profile" />
          <meta property="og:title" content={this.props.name} />
          <meta property="og:description" content={this.props.description} />
          <meta property="og:image" content={this.props.image} />
        </Helmet>
        <div className="container">
          <section className="artist">
            <div
              className="artist__header"
              style={{
                backgroundImage: `url(${this.props.headerImage}`
              }}
            ></div>
            <div className="artist__info">
              <div className="artist__block">
                <img
                  src={this.props.image}
                  alt={this.props.name}
                  className="artist__image"
                />
                <h2 className="artist__name">{this.props.name}</h2>
                <ul className="artist__socials">
                  {Object.entries(this.props.socials).map(
                    ([name, userName]) => {
                      if (!userName) {
                        return ''
                      }

                      return (
                        <Social
                          key={name}
                          data={socialsMap[name].data}
                          isLooped={socialsMap[name].isLooped}
                          link={`https://${name}.com/${userName}`}
                        />
                      )
                    }
                  )}
                </ul>
                <div className="artist__description-wrapper">
                  <div className="artist__description">
                    {this.props.description.split('\n').map((item, index) => {
                      return <p key={index}>{item}</p>
                    })}
                  </div>
                </div>
                {this.props.video && (
                  <Block title="Music Video">
                    <iframe
                      className="video"
                      title="Clip"
                      width="560"
                      height="315"
                      src={`${this.props.video}?autoplay=1&mute=1`}
                      frameBorder="0"
                      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
                      allowFullScreen
                    ></iframe>
                  </Block>
                )}
              </div>
              <div className="artist__block">
                {this.props.albums.length > 0 && (
                  <Block title="Popular songs">
                    <Songs
                      songs={this.props.albums.slice(0, 3).map(album => ({
                        ...album.songs[0],
                        image: album.image
                      }))}
                      artist={this.props.name}
                    />
                  </Block>
                )}
                <Albums albums={this.props.albums} />
              </div>
            </div>
          </section>
        </div>
      </React.Fragment>
    )
  }
}

const mapStateToProps = state => ({
  ...state.artist
})

const mapDispatchToProps = {
  clearArtists: actions.clearArtists,
  initArtist: actions.initArtist
}

export default connect(mapStateToProps, mapDispatchToProps)(Artist)
