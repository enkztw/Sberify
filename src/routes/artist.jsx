import React from 'react'

import { connect } from 'react-redux'
import { actions } from '../store'

import Social from '../components/social'
import Songs from '../components/songs'
import Albums from '../components/albums'
import Video from '../components/video'
import Nav from '../components/nav'

import twitter from '../lottie/twitter'
import facebook from '../lottie/facebook'
import instagram from '../lottie/instagram'

import '../App.scss'

const socialsMap = {
  twitter: {
    isLooped: true,
    data: twitter
  },
  facebook: {
    isLooped: true,
    data: facebook
  },
  instagram: {
    isLooped: false,
    data: instagram
  }
}

class Artist extends React.Component {
  constructor(props) {
    super(props)

    this.props.initArtist(this.props.match.params.name)
  }

  render() {
    return (
      <div className="container">
        <section className="artist">
          <div
            className="artist__header"
            style={{
              backgroundImage: `url(${this.props.headerImage}`
            }}
          ></div>
          <div className="artist__info">
            <div className="artist__block">
              <img
                src={this.props.image}
                alt={this.props.name}
                className="artist__image"
              />
              <h2 className="artist__name">{this.props.name}</h2>
              <ul className="artist__socials">
                {Object.entries(this.props.socials).map(([name, userName]) => {
                  if (!userName) {
                    return ''
                  }

                  return (
                    <Social
                      key={name}
                      data={socialsMap[name].data}
                      isLooped={socialsMap[name].isLooped}
                      link={`https://${name}.com/${userName}`}
                    />
                  )
                })}
              </ul>
              <div className="artist__description-wrapper">
                <div className="artist__description">
                  {this.props.description.split('\n').map((item, index) => {
                    return <p key={index}>{item}</p>
                  })}
                </div>
              </div>
              {this.props.video && <Video link={this.props.video} />}
            </div>
            <div className="artist__block">
              {this.props.albums.length > 0 && (
                <Songs
                  title="Popular songs"
                  songs={this.props.albums.slice(0, 3).map(album => ({
                    ...album.songs[0],
                    image: album.image
                  }))}
                  artist={this.props.name}
                />
              )}
              <Albums albums={this.props.albums} />
            </div>
          </div>
        </section>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  ...state.artist
})

const mapDispatchToProps = {
  initArtist: actions.initArtist
}

export default connect(mapStateToProps, mapDispatchToProps)(Artist)
