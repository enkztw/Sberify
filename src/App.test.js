import React from 'react'
import ReactDOM from 'react-dom'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import thunkMiddleware from 'redux-thunk'
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'

import { shallow } from 'enzyme'
import { configure } from 'enzyme'

import App from './App'
import Home from './routes/home'
import { Artist } from './routes/artist'
import mockData from './mock-data'

const mockStore = configureStore([thunkMiddleware])
let store

beforeEach(() => {
  store = mockStore(initialState)
})

const initialState = {
  app: {
    searchResults: [],
    isLoading: false,
    redirectUri: '',
    error: null
  }
}

configure({ adapter: new Adapter() })

it('Renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
  ReactDOM.unmountComponentAtNode(div)
})

test('Searchmode change on click', () => {
  const home = renderer.create(<Home store={store} />)
  const homeInstance = home.root
  const search = homeInstance.findByProps({ type: 'text' })

  let tree = home.toJSON()
  expect(tree).toMatchSnapshot()

  search.props.onFocus()

  tree = home.toJSON()
  expect(tree).toMatchSnapshot()

  search.props.onBlur()

  tree = home.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Search artists', () => {
  const mock = new MockAdapter(axios)

  mock.onGet('/api/artist/Architects').reply(
    200,
    mockData.find(artist => artist.name === 'Architects')
  )

  axios.get('/api/artist/Architects').then(response => {
    expect('Architects').toEqual(response.data.name)
  })
})

test('Artist renders without crashing', () => {
  const artist = shallow(
    <Artist
      {...mockData[0]}
      clearArtists={() => console.log('Clear')}
      initArtist={() => console.log('Init')}
      match={{
        params: {
          name: 'Architects'
        }
      }}
    />
  )

  expect(artist.find('.artist__name').text()).toEqual('Architects')
})
