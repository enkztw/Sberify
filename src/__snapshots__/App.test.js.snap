// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchmode change on click 1`] = `
<div
  className="container"
>
  <section
    className="home"
  >
    <div
      className="home__header"
    >
      <div
        className="home__logo-wrap"
      >
        <img
          alt="Sberify Music App"
          className="home__logo"
          src="logo.svg"
        />
      </div>
    </div>
    <form
      className="home__search"
      onSubmit={[Function]}
    >
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <p
        className="home__message home__message--add"
      >
        Type the name of the artist 
        <span>
          (GeniusAPI)
        </span>
      </p>
      <p
        className="home__message home__message--find"
        onMouseDown={[Function]}
      >
        Can't find? Add your own
      </p>
      <p
        className="home__message home__message--error"
      />
      <button
        className="home__search-button"
        onMouseDown={[Function]}
        type="button"
      >
        <div
          aria-label="animation"
          role="button"
          style={
            Object {
              "height": "100%",
              "margin": "0 auto",
              "outline": "none",
              "overflow": "hidden",
              "width": "100%",
            }
          }
          tabIndex="0"
          title=""
        />
      </button>
      <div
        className="dropdown-menu-wrapper"
      >
        <ul
          className="dropdown-menu"
        />
      </div>
    </form>
  </section>
</div>
`;

exports[`Searchmode change on click 2`] = `
<div
  className="container"
>
  <section
    className="home"
  >
    <div
      className="home__header"
    >
      <div
        className="home__logo-wrap"
      >
        <img
          alt="Sberify Music App"
          className="home__logo"
          src="logo.svg"
        />
      </div>
    </div>
    <form
      className="home__search home__search--focused"
      onSubmit={[Function]}
    >
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <p
        className="home__message home__message--add"
      >
        Type the name of the artist 
        <span>
          (GeniusAPI)
        </span>
      </p>
      <p
        className="home__message home__message--find"
        onMouseDown={[Function]}
      >
        Can't find? Add your own
      </p>
      <p
        className="home__message home__message--error"
      />
      <button
        className="home__search-button"
        onMouseDown={[Function]}
        type="button"
      >
        <div
          aria-label="animation"
          role="button"
          style={
            Object {
              "height": "100%",
              "margin": "0 auto",
              "outline": "none",
              "overflow": "hidden",
              "width": "100%",
            }
          }
          tabIndex="0"
          title=""
        />
      </button>
      <div
        className="dropdown-menu-wrapper"
      >
        <ul
          className="dropdown-menu"
        />
      </div>
    </form>
  </section>
</div>
`;

exports[`Searchmode change on click 3`] = `
<div
  className="container"
>
  <section
    className="home"
  >
    <div
      className="home__header"
    >
      <div
        className="home__logo-wrap"
      >
        <img
          alt="Sberify Music App"
          className="home__logo"
          src="logo.svg"
        />
      </div>
    </div>
    <form
      className="home__search"
      onSubmit={[Function]}
    >
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <p
        className="home__message home__message--add"
      >
        Type the name of the artist 
        <span>
          (GeniusAPI)
        </span>
      </p>
      <p
        className="home__message home__message--find"
        onMouseDown={[Function]}
      >
        Can't find? Add your own
      </p>
      <p
        className="home__message home__message--error"
      />
      <button
        className="home__search-button"
        onMouseDown={[Function]}
        type="button"
      >
        <div
          aria-label="animation"
          role="button"
          style={
            Object {
              "height": "100%",
              "margin": "0 auto",
              "outline": "none",
              "overflow": "hidden",
              "width": "100%",
            }
          }
          tabIndex="0"
          title=""
        />
      </button>
      <div
        className="dropdown-menu-wrapper"
      >
        <ul
          className="dropdown-menu"
        />
      </div>
    </form>
  </section>
</div>
`;
